pkg/reconciler/taskrun/resources/apply.go

```diff
@@ -165,8 +175,9 @@ func replacementsArrayIdxStepResults(step *v1.Step, paramName string, stepName s
 func replacementsFromStepResults(step *v1.Step, stepParams v1.Params, defaults []v1.ParamSpec) (map[string]string, error) {
 	stringReplacements := map[string]string{}
 	for _, sp := range stepParams {
```

Comment:
If `sp` is nil this could lead to a panic. Should we add an assertion to prevent this or is it guaranteed that `sp` will always be valid?

Reply:
I checked the trail, we can guarantee that `sp` will be valid. Also, it's a value and not a pointer so we're good.